test/reducers/campaigns.spec.js

```diff
@@ -0,0 +1,104 @@
+import deepFreeze from 'deep-freeze';
+import campaigns from '../../app/assets/javascripts/reducers/campaigns';
+import {
+  RECEIVE_COURSE_CAMPAIGNS,
+  RECEIVE_ALL_CAMPAIGNS,
+  ADD_CAMPAIGN,
+  SORT_CAMPAIGNS_WITH_STATS,
+  SORT_ALL_CAMPAIGNS
+} from '../../app/assets/javascripts/constants';
+
+const campaignsArray = [
+  { id: 2, slug: 'second_campaign', description: 'second description' },
+  { id: 1, slug: 'first_campaign', description: 'first description' }
+];
+const sortedcampaignsArray = [
+  { id: 1, slug: 'first_campaign', description: 'first description' },
+  { id: 2, slug: 'second_campaign', description: 'second description' },
+];
+
+describe('campaigns reducer', () => {
+  test('should return initial state when no action nor state is provided', () => {
+    const newState = campaigns(undefined, { type: null });
+    expect(newState.campaigns).toEqual([]);
+    expect(newState.all_campaigns).toEqual([]);
+    expect(newState.isLoaded).toBe(false);
+  });
+
+  test('should add and return campaigns data with ADD_CAMPAIGN and set isLoaded to true', () => {
+    const initialState = { campaigns: [], all_campaigns: [], isLoaded: false };
+    deepFreeze(initialState);
+
+
+    // Review needed: The implementation of the ADD_CAMPAIGN case in the reducer
+    // is unclear regarding whether it should replace the existing campaigns or
+    // append to them. The expected behavior of the state update should be clarified.
```

Comment:
The intended behavior is to return the up-to-date campaigns list in response to adding a new campaign, so replacing the existing campaigns is the expected behavior.

Reply:
I understand now, @ragesoss  is this ready for or i should remove this comment

Reply:
Please remove the comment.

Reply:
@ragesoss I have removed it